orangewidget/widget.py

```diff
@@ -332,45 +334,106 @@ def __new__(cls, *args, captionTitle=None, **kwargs):
+        self.__save_image_action.setVisible(bool(self.graph_name))
 
+        self.__reset_action = QAction(
+            "Reset", self, objectName="action-reset-settings",
+            toolTip="Reset settings to default state",
+            enabled=False, visible=False,
+        )
         if hasattr(self, "reset_settings"):
-            self.__reset_action = QAction(
-                "Reset", self, objectName="action-reset-settings",
-                toolTip="Reset settings to default state",)
             self.__reset_action.triggered.connect(self.reset_settings)
+            self.__reset_action.setEnabled(True)
+            self.__reset_action.setVisible(True)
 
+        self.__visual_settings_action = QAction(
+            "Show View Options", self, objectName="action-visual-settings",
+            toolTip="Show View Options",
+            enabled=False, visible=False,
+        )
+        self.__visual_settings_action.triggered.connect(
+            self.openVisualSettingsClicked)
         if hasattr(self, "set_visual_settings"):
-            assert self.initial_visual_settings is not None
-            self.__visual_settings_action = QAction(
-                "Show View Options", self, objectName="action-visual-settings",
-                toolTip="Show View Options",)
-            self.__visual_settings_action.triggered.connect(
-                self.openVisualSettingsClicked)
+            self.__visual_settings_action.setEnabled(True)
+            self.__visual_settings_action.setVisible(True)
 
         self.addAction(self.__help_action)
 
-        self.__info_ns = None
+        self.__copy_action = QAction(
+            "Copy to Clipboard", self, objectName="action-copy-to-clipboard",
+            shortcut=QKeySequence.Copy, enabled=False, visible=False
+        )
+        self.__copy_action.triggered.connect(self.copy_to_clipboard)
+        if type(self).copy_to_clipboard != OWBaseWidget.copy_to_clipboard \
+                or self.graph_name is not None:
+            self.__copy_action.setEnabled(True)
+            self.__copy_action.setVisible(True)
+            self.__copy_action.setText("Copy Image to Clipboard")
+
+        # macOS Minimize action
+        self.__minimize_action = QAction(
+            "Minimize", self, shortcut=QKeySequence("ctrl+m")
+        )
+        self.__minimize_action.triggered.connect(self.showMinimized)
+        # macOS Close window action
+        self.__close_action = QAction(
+            "Close", self, objectName="action-close-window",
+            shortcut=QKeySequence("ctrl+w")
+        )
+        self.__close_action.triggered.connect(self.hide)
+
+        self.__menubar = mb = QMenuBar(self)
+        fileaction = mb.addMenu(_Menu("&File", mb, objectName="menu-file"))
+        fileaction.setVisible(False)
+        fileaction.menu().addSeparator()
+        fileaction.menu().addAction(self.__report_action)
+        fileaction.menu().addAction(self.__save_image_action)
```

Comment:
`__reset_action` is not added to any menu. If this is intentional, just confirm and I'll be OK with it.

Reply:
It's added now.